crates/bevy_math/src/curve/mod.rs

```diff
@@ -968,19 +971,135 @@ where
     }
 }
 
+/// Convert a `start` and `end` point of some [`VectorSpace`] type into a [`Curve`] over the
+/// [unit interval], sampled by linearly interpolating between the two boundary values.
+///
+/// [unit interval]: `Interval::UNIT`
+pub fn line_curve<T: VectorSpace>(start: T, end: T) -> impl Curve<T> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: move |t: f32| start * (1.0 - t) + end * t,
```

Comment:
Are closures the most effective tool (taking into account perf) to use for all these curves, would a custom struct implementing Curve be somehow better/worse?